packages/memory/reference.ts

```diff
@@ -5,4 +5,44 @@ export * from "merkle-reference";
 // workaround it like this.
 export const fromString = Reference.fromString as (
   source: string,
-) => Reference.Reference;
+) => Reference.View;
+
+/**
+ * Bounded LRU cache for memoizing refer() results.
+ * refer() is a pure function (same input → same output), so caching is safe.
+ * We use JSON.stringify as the cache key since it's ~25x faster than refer().
+ */
+const CACHE_MAX_SIZE = 1000;
```

Comment:
this is probably trivial in memory consumption relative other things, so we could easily increase it. but i also have no intuition whether that would actually make a real difference. might be interesting to monitor the hit rate (super not urgent, just a thought)